docs/imagecustomizer/concepts/imagehistory.md

```diff
@@ -0,0 +1,136 @@
+# Image History
+
+The Image History feature records the customization history of images in a
+structured JSON format. It enables users to:
+
+- Trace the lineage of customized images.
+- Verify input integrity using SHA256 hashes.
+- Debug and identify issues in customization workflows.
+
+Image history tracking can be disabled by explicitly configuring the option in
+the [OS configuration settings](../api/configuration/os.md)
+
+## History Storage
+
+- **Location:** `/usr/share/image-customizer/history.json`.
+- **Format:** An array of JSON objects, each representing a customization step.
+- **Order:** Each new history entry is appended to the end of the existing list
+  within the history file, preserving chronological sequence across image
+  customizations.
+
+## Metadata Captured
+
+Each entry in the history includes:
+
+- **Timestamp:** Time of customization.
+- **Tool Version:** Image customizer version used.
+- **Image UUID:** Unique identifier for the image.
+- **Configuration:** Captures the config used for customization (with certain
+  modifications).
+
+## Config Modifications
+
+The configuration undergoes specific modifications before being serialized into
+JSON format:
+
+**Hashing:**
+
+- SHA256 hashes are generated for scripts, additional files, and the contents of
+  additional directories to ensure input integrity.
+- If only content is provided for scripts or additional files (and no source
```

Comment:
do you mean 'inlined'? For scripts, users can inline the script in the yaml config file... is there something similar to files?

Reply:
yes, like 'content' field instead of the path to the file, essentially inline